src/main/java/ch/njol/skript/effects/EffToggle.java

```diff
@@ -71,15 +78,18 @@
 		setDataMethod = mh;
 	}
 	
-	@SuppressWarnings("null")
-	private Expression<Block> blocks;
+	private Expression<?> expressions;
 	private int toggle;
 	
-	@SuppressWarnings({"unchecked", "null"})
 	@Override
 	public boolean init(final Expression<?>[] vars, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		blocks = (Expression<Block>) vars[0];
+		expressions = (Expression<?>) vars[0];
 		toggle = matchedPattern - 1;
+		Object type = expressions.getReturnType();
+		if (type instanceof Boolean && !CollectionUtils.containsSuperclass(expressions.acceptChange(ChangeMode.SET), Boolean.class)) {
```

Comment:
getReturnType returns Class, and calling `x instanceof Boolean` with x a Class is always false.
```suggestion
		if (expressions.getReturnType() == Boolean.class && !CollectionUtils.containsSuperclass(expressions.acceptChange(ChangeMode.SET), Boolean.class)) {
```

Reply:
Shouldn't this check for acceptChange with Boolean[].class if the expression is plural?